src/commands/warn.ts

```diff
@@ -0,0 +1,78 @@
+import { CommandInteraction } from "discord.js";
+import { SlashCommandBuilder, userMention, hideLinkEmbed } from "@discordjs/builders";
+import { getTexts } from "../common";
+
+const CONDUCT = "conduct";
+const CONTENT = "content";
+const SPAM = "spam";
+type Reason = typeof CONDUCT | typeof CONTENT | typeof SPAM;
+
+const reasons = [
+  {
+    name: CONDUCT,
+    description: "Poor conduct: harassment, witch hunting, sexism, etc.",
+  },
+  {
+    name: CONTENT,
+    description: "Offensive / NSFW content: nudity, sex, violence, etc.",
+  },
+  {
+    name: SPAM,
+    description:
+      "Spam / phishing: unauthorized server invites, advertisements, malicious links etc.",
+  },
+];
+
+const warnTexts: Map<string, string> = getTexts(
+  "warn",
+  reasons.map((reason) => reason.name)
+);
+
+// warn
+export const data = new SlashCommandBuilder()
+  .setName("warn")
+  .setDescription("Warn the user for violation of server rules")
+  .setDefaultPermission(false)
+  .addUserOption((option) =>
+    option.setName("user").setDescription("The user who violated server rules").setRequired(true)
+  )
+  .addStringOption((option) =>
+    option
+      .setName("reason")
+      .setDescription("The reason the specified user violated server rules")
+      .setRequired(true)
+      .addChoices(reasons.map((reason) => [reason.description, reason.name]))
+  )
+  .toJSON();
+
+export const authorizedRoles = ["admin", "mods"];
+
+export const call = async (interaction: CommandInteraction) => {
+  const user = interaction.options.getUser("user", true);
+  const reason = interaction.options.getString("reason", true) as Reason;
+  try {
+    const reply = warnTexts.get(reason);
+    if (!reply) {
+      await interaction.reply({
+        content: `Could not get the text for reason "${reason}"
+
+Text is available for the following reasons:
+
+${reasons.map((reason) => `- ${reason.name}`).join("\n")}`,
+        ephemeral: true,
+      });
+      return;
+    }
```

Comment:
Move these lines out of `try`. If sending `ephemeral` message can fail, it should be in a separate `try` because the `catch` block doesn't make sense for it.